benchmark/sqllogictest_tester/run_tester.sh

```diff
@@ -0,0 +1,304 @@
+if [ -z "$CREDS_HASH" ]; then fail Must supply CREDS_HASH; fi
+if [ -z "$TMP_TESTING_DIR" ]; then fail Must supply TMP_TESTING_DIR; fi
+if [ -z "$TMP_CSV_DIR" ]; then fail Must supply TMP_CSV_DIR; fi
+if [ -z "$TEST_FILE_DIR_LIST" ]; then fail Must supply TEST_FILE_DIR_LIST; fi
+if [ -z "$COMMITS_TO_TEST" ]; then fail Must supply COMMITS_TO_TEST; fi
+if [ -z "$DOLT_BRANCH" ]; then fail Must supply DOLT_BRANCH; fi
+if [ -z "$TEST_N_TIMES" ]; then fail Must supply TEST_N_TIMES; fi
+
+[[ "$TEST_N_TIMES" =~ ^[0-9]+$ ]] || fail TEST_N_TIMES must be a number
+
+function setup() {
+    rm -rf "$CREDSDIR"
+    mkdir -p "$CREDSDIR"
+    cat "$DOLT_CREDS" > "$CREDSDIR"/"$CREDS_HASH".jwk
+    echo "$DOLT_GLOBAL_CONFIG" > "$DOLT_CONFIG_PATH"/config_global.json
+    dolt config --global --add user.creds "$CREDS_HASH"
+    dolt config --global --add metrics.disabled true
+    dolt version
+}
+
+function setup_testing_dir() {
+    rm -rf "$TMP_TESTING_DIR"
+    mkdir -p "$TMP_TESTING_DIR"
+
+    IFS=', ' read -r -a test_list <<< "$TEST_FILE_DIR_LIST"
+    for fd in "${test_list[@]}"
+    do
+         dir=$(dirname "$fd")
+         mkdir -p "$TMP_TESTING_DIR"/"$dir"
+         cp -r "$sqllogictest_checkout"/test/"$fd" "$TMP_TESTING_DIR"/"$fd"
+    done
+}
+
+function with_dolt_commit() {
+  local commit_hash="$1"
+    (
+      cd "$base_dir"/tempDolt/go
+      git checkout master
+
+      exists=$(git branch --list "temp-$commit_hash"| sed '/^\s*$/d' | wc -l)
+
+      if [ "$exists" -eq 0 ]; then
+        git checkout -b "temp-$commit_hash" "$commit_hash";
+      else
+        git checkout "temp-$commit_hash";
+      fi
+
+      git log -n 1
+
+      if ! [ -x "$base_dir"/.ci_bin/"$commit_hash"/dolt ]; then
+          if ! [ -d "$base_dir"/.ci_bin ]; then
+            mkdir -p "$base_dir"/.ci_bin/"$commit_hash"
+          fi
+
+          go get -mod=readonly ./...
+          go build -mod=readonly -o "$base_dir"/.ci_bin/"$commit_hash"/dolt ./cmd/dolt/.
+      fi
+    )
+
+    echo "Finished installing dolt from $commit_hash:"
+    export PATH="$base_dir/.ci_bin/$commit_hash":$old_path
+    dolt version
+}
+
+function with_dolt_checkout() {
+    (
+      cd "$base_dir"/go
+      if ! [ -x "$base_dir"/.ci_bin/checkout/dolt ]; then
+          if ! [ -d "$base_dir"/.ci_bin ]; then
+            mkdir -p "$base_dir"/.ci_bin/checkout
+          fi
+          go get -mod=readonly ./...
+          go build -mod=readonly -o "$base_dir"/.ci_bin/checkout/dolt ./cmd/dolt/.
+      fi
+    )
+    echo "Finished installing dolt from checkout:"
+    export PATH="$base_dir/.ci_bin/checkout":$old_path
+    dolt version
+}
+
+function import_once() {
+    local commit_hash="$1"
+    local test_num="$2"
+    local parsed="$log_dir/parsed-$commit_hash-$test_num".json
+
+    dolt checkout "$DOLT_BRANCH"
+
+    exists=$(dolt branch --list "temp-$commit_hash"| sed '/^\s*$/d' | wc -l | tr -d '[:space:]')
+
+    if [ "$exists" -eq 0 ]; then
+      dolt checkout -b "temp-$commit_hash";
```

Comment:
How does this create the right branch? Shouldn't this be `dolt checkout -b "temp-$commit_hash" "$commit_hash"`?